Assets/HotmaxEditorVR/propertiesMenu/propertiesMenu.cs

```diff
@@ -0,0 +1,246 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+
+public class propertiesMenu : MonoBehaviour
+{
+    //-----DEPENDENCIES-----\\
+    //--All materials must be in a Materials folder in the "Resources"
+    //--prefab called "lightSource" must be in the Resources folder
+    //--All imported textures must be read/writable
+
+    //--local refs
+    private GameObject _selectedObject = stateManager.selectedObject;
+    private stateManager.editorModes _editorMode = stateManager.editorMode;
+    private Vector3 _raycastHitPoint;
+    private GameObject _raycastHitObject;
+
+
+    //--add all materials in object to list
+    private List<string> materialNames = new List<string>();
+    private List<List<string>> chunkedMaterialList = new List<List<string>>();
+    private int currentPage = 0;
```

Comment:
if you make variables like this, make them more explicit like currentMaterialPage or something

Reply:
got it dude